src/Features/CSharp/Completion/CompletionProviders/CompletionUtilities.cs

```diff
@@ -81,7 +81,13 @@ internal static bool IsTriggerAfterSpaceOrStartOfWordCharacter(SourceText text,
         {
             // Bring up on space or at the start of a word.
             var ch = text[characterPosition];
-            return ch == ' ' || (CompletionUtilities.IsStartingNewWord(text, characterPosition) && options.GetOption(CompletionOptions.TriggerOnTypingLetters, LanguageNames.CSharp));
+            return SpaceTypedNotAdjacentToWord(ch, text, characterPosition) ||
+                (CompletionUtilities.IsStartingNewWord(text, characterPosition) && options.GetOption(CompletionOptions.TriggerOnTypingLetters, LanguageNames.CSharp));
+        }
+
+        private static bool SpaceTypedNotAdjacentToWord(char ch, SourceText text, int characterPosition)
+        {
+            return ch == ' ' && (characterPosition == text.Length - 1 || !IsWordStartCharacter(text[characterPosition + 1]));
```

Comment:
Should this space "adjacent to a word" as well: "word<space>"? Or should "AdjacentTo" just be "Before"?